src/full/Agda/TypeChecking/Sort.hs

```diff
@@ -85,28 +86,39 @@ hasBiggerSort = void . inferUnivSort
 -- | Infer the sort of a pi type. If we can compute the sort straight away,
 --   return that. Otherwise, return @PiSort a s2@ and add a constraint to
 --   ensure we can compute the sort eventually.
-inferPiSort :: PureTCM m => Dom Type -> Abs Sort -> m Sort
+inferPiSort :: (PureTCM m,MonadConstraint m) => Dom Type -> Abs Sort -> m Sort
 inferPiSort a s2 = do
   s1' <- reduce $ getSort a
-  s2' <- mapAbstraction a reduce s2
+  s2' <- reduce s2
```

Comment:
Calling reduce here is not necessary and in fact a bad idea, since the `Reduce` instance for `Abs` uses a dummy type (this is the reason for using `mapAbstraction` below instead)